rbac/management/tenant_service/v2.py

```diff
@@ -103,6 +105,125 @@ def update_user(
+        existing_custom_default_group_ids = set()
+        for tenant in tenants:
+            kwargs = {"tenant": tenant}
+            if hasattr(tenant, "platform_default_groups") and tenant.platform_default_groups:
+                kwargs["default_group_uuid"] = tenant.platform_default_groups[0].uuid
+                existing_custom_default_group_ids.add(tenant.platform_default_groups[0].uuid)
+            mappings_to_create.append(TenantMapping(**kwargs))
+
+            root_workspace_uuid = uuid4()
+            default_workspace_uuid = uuid4()
+            workspaces.append(
+                Workspace(uuid=root_workspace_uuid, tenant=tenant, type=Workspace.Types.ROOT, name="Root Workspace")
+            )
+            workspaces.append(
+                Workspace(
+                    uuid=default_workspace_uuid, tenant=tenant, type=Workspace.Types.DEFAULT, name="Default Workspace"
+                )
+            )
+            default_workspace_uuids.append(default_workspace_uuid)
+            tenant_id = f"{self._user_domain}/{tenant.org_id}"
+            relationships = [
+                create_relationship(
+                    ("rbac", "workspace"),
+                    str(default_workspace_uuid),
+                    ("rbac", "workspace"),
+                    str(root_workspace_uuid),
+                    "parent",
+                ),
+                create_relationship(
+                    ("rbac", "workspace"), str(root_workspace_uuid), ("rbac", "tenant"), tenant_id, "parent"
+                ),
+            ]
+
+            # Include platform for tenant
+            relationships.append(
+                create_relationship(("rbac", "tenant"), tenant_id, ("rbac", "platform"), settings.ENV_NAME, "platform")
+            )
+
+        Workspace.objects.bulk_create(workspaces)
+
+        mappings = TenantMapping.objects.bulk_create(mappings_to_create)
+        tenant_mappings = {mapping.tenant_id: mapping for mapping in mappings}
+        bootstrapped_tenants = []
+
+        for tenant, default_workspace_uuid in zip(tenants, default_workspace_uuids):
+            mapping = tenant_mappings[tenant.id]
+            relationships.extend(
+                self._bootstrap_default_access(
+                    tenant, mapping, str(default_workspace_uuid), existing_custom_default_group_ids
+                )
+            )
+            bootstrapped_tenants.append(BootstrappedTenant(tenant, mapping))
+        self._replicator.replicate(
+            ReplicationEvent(
+                event_type=ReplicationEventType.BOOTSTRAP_TENANT,
+                info={"CJI": "bulk_bootstrap"},
+                partition_key="rbactodo",
+                add=relationships,
+            )
+        )
+        return bootstrapped_tenants
+
+    def _get_or_bootstrap_tenants(self, org_ids: set) -> list[BootstrappedTenant]:
+        """Bootstrap list of tenants, used by update_users."""
+        # Fetch existing tenants
+        existing_tenants = {
```

Comment:
can this be simplified like this ?
```
existing_tenants = {
    tenant.org_id: tenant
    for tenant in Tenant.objects.filter(org_id__in=org_ids)
    .select_related("tenant_mapping")
    .prefetch_related(
        Prefetch(
            "group_set",
            queryset=Group.objects.filter(platform_default=True),
            to_attr="platform_default_groups",
        ),
        Prefetch(
            "workspace_set",
            queryset=Workspace.objects.filter(type__in=[Workspace.Types.ROOT, Workspace.Types.DEFAULT]),
            to_attr="workspaces",
        )
    )
}
```

Reply:
Oh, I missed this one. Thanks! I update the logic to consider not bootstrapped ones by the tenant mapping, so the workspaces are not used anymore. Removed